pkg/operator/ceph/object/spec.go

```diff
@@ -318,37 +329,27 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) v1.Container {
 		return v1.Container{}
 	}
 	if kmsEnabled {
-		container.Args = append(container.Args,
-			cephconfig.NewFlag("rgw crypt s3 kms backend",
-				c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.Provider]),
-			cephconfig.NewFlag("rgw crypt vault addr",
-				c.store.Spec.Security.KeyManagementService.ConnectionDetails[api.EnvVaultAddress]),
-		)
+		if c.store.Spec.Security.IsRGWSSEKMSEnabled() {
+			container.Args = append(container.Args, c.setRGWOptions(rgwSSEKMSDefaultOptions)...)
+		}
+		if c.store.Spec.Security.IsRGWSSES3Enabled() {
+			container.Args = append(container.Args, c.setRGWOptions(rgwSSES3DefaultOptions)...)
+		}
 		if c.store.Spec.Security.KeyManagementService.IsTokenAuthEnabled() {
-			container.Args = append(container.Args,
-				cephconfig.NewFlag("rgw crypt vault auth", kms.KMSTokenSecretNameKey),
-				cephconfig.NewFlag("rgw crypt vault token file",
-					path.Join(rgwVaultDirName, kms.VaultFileName)),
-				cephconfig.NewFlag("rgw crypt vault prefix", c.vaultPrefixRGW()),
-				cephconfig.NewFlag("rgw crypt vault secret engine",
-					c.store.Spec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]),
-			)
+			if c.store.Spec.Security.IsRGWSSEKMSEnabled() {
+				container.Args = append(container.Args, c.setRGWOptions(rgwSSEKMSVaultTokenOptions)...)
```

Comment:
Instead of a single method with a switch statement, how about a separate method for each case? It really will make the code easier to read if the function just directly returns the options, and no need to define consts for the switch statement. Something like this:
```suggestion
				container.Args = append(container.Args, kmsVaultTokenOptions()...)
```